### PR TITLE
Update dependencies to include D3

### PR DESCRIPTION
**Pull request by LujieZ**
This PR updates the dependencies to include the D3 library for the visual 
elements that we will be creating for one-push and all-pushes pages. The 
D3.js library [1] is a JavaScript library for visualizing data using web standards.

This PR also removes dependencies that are declared twice in `package.json`
and `package-lock.json`.

[1] https://github.com/d3/d3

### CODE REVIEW

**Comment by yinggao** [comment]
> Can we make the list sorted in alphabetical order?

File: web/package.json [hunk 1]
```diff
@@ -41,9 +43,6 @@
     "@babel/core": "^7.0.0",
     "@babel/preset-env": "^7.11.0",
     "@babel/preset-react": "^7.10.4",
-    "@types/jasmine": "~3.5.0",
-    "@types/jasminewd2": "~2.0.3",
-    "@types/node": "^12.11.1",
```
Comment (razvanm) [inline web/package.json:46]: Why it's safe to remove these dependencies?

Comment (LujieZ) [reply web/package.json:46]: I think they are removed when I did `npm install`. They are added back now!

Comment (yinggao) [reply web/package.json:46]: Why they've moved to the 'dependencies' section from the 'devDependencies' section then?

Comment (LujieZ) [reply web/package.json:46]: I think I just made the same mistake...

Comment (yinggao) [reply web/package.json:46]: Instead of removing the duplicates here, can we remove the @type/xxx from above to keep the list sorted?

**Pull request merged**